src/loki/search.cc

```diff
@@ -35,15 +35,12 @@ bool is_search_filter_triggered(const DirectedEdge* edge,
   // Note that min_ and max_road_class are integers where, by default, max_road_class
   // is 0 and min_road_class is 7. This filter rejects roads where the functional
   // road class is outside of the min to max range.
-  if ((road_class > min_road_class || road_class < max_road_class) ||
-      (search_filter.exclude_tunnel_ && edge->tunnel()) ||
-      (search_filter.exclude_bridge_ && edge->bridge()) ||
-      (search_filter.exclude_ramp_ && (edge->use() == Use::kRamp)) ||
-      (search_filter.exclude_closures_ && costing.IsClosed(edge, tile))) {
-    return true;
-  }
-
-  return false;
+  return (road_class > min_road_class || road_class < max_road_class) ||
+         (search_filter.exclude_tunnel_ && edge->tunnel()) ||
+         (search_filter.exclude_bridge_ && edge->bridge()) ||
+         (search_filter.exclude_ramp_ && (edge->use() == Use::kRamp)) ||
+         (search_filter.exclude_closures_ && (costing.flow_mask() & kCurrentFlowMask) &&
+          tile->IsClosed(edge));
```

Comment:
im not sure why there was an `if` here, its gone now. also i reverted the changes to calling isclosed directly on the costing. i didnt like that it hides state from us. instead i made sure that `exclude_closures` is always properly set at request parsing time so we can rely on it there (wther the user set it or it was overriden by ignore_closures in the costing options)